rice-middleware/core/impl/src/main/java/org/kuali/rice/core/impl/datetime/DateTimeServiceImpl.java

```diff
@@ -233,27 +233,20 @@ public int dateDiff(Date startDate, Date endDate, boolean inclusive) {
 		Calendar endDateCalendar = Calendar.getInstance();
 		endDateCalendar.setTime(endDate);
 
-		int startDateOffset = -(startDateCalendar.get(Calendar.ZONE_OFFSET) + startDateCalendar
-				.get(Calendar.DST_OFFSET))
-				/ (60 * 1000);
+		int startCalendarOffset = startDateCalendar.get(Calendar.ZONE_OFFSET) + startDateCalendar.get(Calendar.DST_OFFSET);
+		int endCalendarOffset = endDateCalendar.get(Calendar.ZONE_OFFSET) + endDateCalendar.get(Calendar.DST_OFFSET);
 
-		int endDateOffset = -(endDateCalendar.get(Calendar.ZONE_OFFSET) + endDateCalendar
-				.get(Calendar.DST_OFFSET))
-				/ (60 * 1000);
-
-		if (startDateOffset > endDateOffset) {
-			startDateCalendar.add(Calendar.MINUTE, endDateOffset
-					- startDateOffset);
-		}
-
-		if (inclusive) {
-			startDateCalendar.add(Calendar.DATE, -1);
-		}
+		startDateCalendar.add(Calendar.MILLISECOND, startCalendarOffset);
+		endDateCalendar.add(Calendar.MILLISECOND, endCalendarOffset);
```

Comment:
Much cleaner! But not sure what's happening here. According to JavaDocs, Calendar.MILLISECONDS is 
> Field number for get and set indicating the millisecond within the second. E.g., at 10:04:15.250 PM the MILLISECOND is 250.

How does that interact with `.add()`?